RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -17,58 +21,83 @@
 @RequiredArgsConstructor
 @Slf4j
 public class LuceneDocumentsReader {
+    public static Function<Path, LuceneDocumentsReader> getFactory(boolean softDeletesPossible, String softDeletesField) {
+        return path -> new LuceneDocumentsReader(path, softDeletesPossible, softDeletesField);
+    }
     public static final int NUM_DOCUMENTS_BUFFERED = 1024;
+
     protected final Path indexDirectoryPath;
+    protected final boolean softDeletesPossible;
+    protected final String softDeletesField;
 
     public Flux<Document> readDocuments() {
```

Comment:
Need a higher-level comment explaining the various states the documents can be in, and how they'll be handled.